lib/spack/spack/test/util/spack_yaml.py

```diff
@@ -0,0 +1,70 @@
+##############################################################################
+# Copyright (c) 2013-2018, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+import re
+
+import spack.config
+from spack.main import SpackCommand
+
+config_cmd = SpackCommand('config')
+
+
+def get_config_line(pattern, lines):
+    """Get a configuration line that matches a particular pattern."""
+    line = next((l for l in lines if re.search(pattern, l)), None)
+    assert line is not None, 'no such line!'
+    return line
+
+
+def check_blame(element, scope=None):
+    """Get an element from a config file and check its blame information.
+
+    Args:
+        element (str): name of element (e.g. install_tree)
+        scope (str): expected config scope to find it in, if not a file
+    """
+    output = config_cmd('blame', 'config')
+    blame_lines = output.rstrip().split('\n')
+
+    element_line = get_config_line(element + ':', blame_lines)
+    element = spack.config.get('config:%s' % element)
+
+    if scope is None:
+        annotation = '%s:%d' % (element._start_mark.name,
```

Comment:
IMO this is somewhat roundabout since the test is using the output generated by the test to verify the output (e.g. constructing the annotation by accessing the parsed element). I imagine it would be fragile to make an assertion about a specific line number but could a regex be generated to verify it:

`"{0}:\d\s+{1}".format(expected_section_name, cfg_var_name)`

(and you can just use parameters to the check function vs. extracting from the element)

Also: what line number *is* this supposed to be in `test_config_blame_with_override`? I thought the `override` scope doesn't have an associated file; if I'm right about that, would it make sense to test this in one case where there is an associated file?